tower-balance/src/choose/p2c.rs

```diff
@@ -18,13 +18,19 @@ use choose::{Choose, Replicas};
 ///
 /// [finagle]: https://twitter.github.io/finagle/guide/Clients.html#power-of-two-choices-p2c-least-loaded
 /// [p2c]: http://www.eecs.harvard.edu/~michaelm/postscripts/handbook2001.pdf
-#[derive(Debug)]
-pub struct PowerOfTwoChoices<R> {
+#[derive(Debug, Default)]
+pub struct PowerOfTwoChoices<R: Rng = ThreadRng> {
```

Comment:
I am not sure why we need to parameterize the RNG. I think just using one default one would be OK.

If you're going to use the `rand` crate and constructing `PowerOfTwoChoices` objects is infrequently then I would suggest that `PowerOfTwoChoices` just holds a `SmallRng` that it seeds when it is first constructed and never reseeds.

I suggest documenting that the RNG doesn't need to be cryptographically secure but does need to be fast. (This is why I don't like `ThreadRng`; it tries to mostly be cryptographically secure and fast at the same time.)

Reply:
> I am not sure why we need to parameterize the RNG. I think just using one default one would be OK.

OIC, so it can be swapped out for tests, as in the initial description of the PR. AFAICT you don't need to actually parameterize on the choice of PRNG algorithm, but instead you just need to provide a way to override the random seed of the PRNG. For example, if you just used `SmallRng` then you'd just need a constructor that takes a `&[u8; 16]` parameter that you can pass to `SmallRng::from_seed()`.